src/vm/assembly.cpp

```diff
@@ -1965,6 +1965,10 @@ INT32 Assembly::ExecuteMainMethod(PTRARRAYREF *stringArgs, BOOL waitForOtherThre
     IfFailThrow(hr);
     
     END_ENTRYPOINT_THROWS;
+
+    // Sync the latched exit code with the return value of main function.
+    SetLatchedExitCode(iRetVal);
```

Comment:
> Seems like we should remove the check

Agree.

Could you please also verify that the following program will have exit code 123?

```
using System;

class My {
    static void Main() {
        Environment.ExitCode = 123;
    }
}
```